### PR TITLE
fix(StyledOuterWrapper): Fix height and label alignment problems

### LINKED ISSUE #2889: Look at 1px label movement caused by outer wrapper update
Recent changes to the `StyledOuterWrapper` mean that elements have moved by 1px. This needs reviewing with @k9dh3zij and updating affected elements.

[Chromatic example](https://www.chromatic.com/build?appId=5e25c277526d380020b5e418&number=4326)

This issue needs to include resolving issues around label align across all browsers.

## Label alignment issues
image
image

### PR DESCRIPTION
**Pull request by jpveooys**
## Related issue

Resolves #2889 

## Overview

This fixes a few styling problems with the experimental components:

- corrects their heights
- fixes problems with label positioning

## Link to preview

[Storybook](https://red-playground.netlify.app/)

## Reason

To make sure the components look as intended.

## Work carried out

- [x] Rework borders to use box shadows instead
- [x] Update positioning of SelectE options
- [x] Correct heights of components

## Screenshot

A couple of examples:

#### Before

image

#### After

image

#### Before

image

#### After

image

## Developer notes

This was done by making more use of box shadows instead of borders.

Affected components are:

- SelectE
- DatePickerE
- NumberInputE
- TextAreaE
- TextInputE

### CODE REVIEW

File: packages/react-component-library/src/components/SelectE/partials/StyledOptions.tsx [hunk 1]
```diff
@@ -9,13 +9,6 @@ export const StyledOptions = styled.ul`
...
-  border-radius: 0 0
-    calc(
-      ${BORDER_RADIUS[COMPONENT_SIZE.FORMS]} -
-        ${BORDER_WIDTH[COMPONENT_SIZE.FORMS]}
-    )
-    calc(
-      ${BORDER_RADIUS[COMPONENT_SIZE.FORMS]} -
-        ${BORDER_WIDTH[COMPONENT_SIZE.FORMS]}
-    );
+  border-radius: 0 0 calc(${BORDER_RADIUS[COMPONENT_SIZE.FORMS]})
```
Comment (thyhjwb6) [inline packages/react-component-library/src/components/SelectE/partials/StyledOptions.tsx]: This doesn't seem to need the `calc` anymore

**Review by thyhjwb6 — COMMENTED** [review]
Looks great. This is a much cleaner solution.

I noticed something in FireFox:

Screenshot 2021-12-16 at 15 30 59

**Review by thyhjwb6 — APPROVED** [review]
Tested in FF using the link and looks good.

👍

**Comment by k9dh3zij** [comment]
Thanks @jpveooys, looks perfect!! Only one tiny border missing:

NumberInputE disabled...
image

...should look like DatePicker disabled:
image

👍 That's all!

**Comment by jpveooys** [comment]
Thanks @k9dh3zij :smile: 

I've just pushed a change to add that border in – you can see it here: https://red-playground.netlify.app/?path=/docs/number-input-experimental--disabled

**Comment by k9dh3zij** [comment]
👍 Perfect!

**Review by m7kvqbe1 — APPROVED** [review]
LGTM

**Pull request merged**